base/linalg/lapack.jl

```diff
@@ -3970,10 +3996,36 @@ for (trsen, tgsen, elty) in
     end
 end
 
-for (trsen, tgsen, elty) in
-    ((:ztrsen_, :ztgsen_, :Complex128),
-     (:ctrsen_, :ctgsen_, :Complex64))
+for (trexc, trsen, tgsen, elty) in
+    ((:ztrexc_, :ztrsen_, :ztgsen_, :Complex128),
+     (:ctrexc_, :ctrsen_, :ctgsen_, :Complex64))
     @eval begin
+        function trexc!(ifst::BlasInt, ilst::BlasInt, T::StridedMatrix{$elty}, Q::StridedMatrix{$elty})
+# *     .. Scalar Arguments ..
+#       CHARACTER          COMPQ
+#       INTEGER            IFST, ILST, INFO, LDQ, LDT, N
+# *     ..
+# *     .. Array Arguments ..
+#       DOUBLE PRECISION   Q( LDQ, * ), T( LDT, * ), WORK( * )
+            chkstride1(T, Q)
+            n = chksquare(T)
+            ldt = max(1, stride(T, 2))
+            ldq = max(1, stride(Q, 2))
+            work = Array($elty, 1)
+            info = Array(BlasInt, 1)
+
+            ccall(($(blasfunc(trexc)), liblapack), Void,
+                  (Ptr{UInt8},  Ptr{BlasInt},
+                   Ptr{$elty}, Ptr{BlasInt}, Ptr{$elty}, Ptr{BlasInt},
+                   Ptr{BlasInt}, Ptr{BlasInt},
+                   Ptr{$elty}, Ptr{BlasInt}),
+                  &'V', &n,
```

Comment:
Usually, we don't restrict the options supported by LAPACK so I'd prefer a version with a `compq` argument. I'd be fine with having a second method without the `compq` argument that calls the first with `compq=='V'`.